### DISCUSSION TITLE
integration with worker mode of FrankenPHP ?

### DISCUSSION THREAD

**Discussion by abda11ah** [mikecao/flight#612] [General]
I saw in a video that FrankenPHP was an incredible way to optimize PHP code execution. The worker mode keeps the PHP binary loaded in RAM to respond to requests very quickly. However, it is said that this is only compatible with Laravel and Symfony currently. Does a similar method exist within FlightPHP or at least a sample code that i can add to my index.php file ?

**Comment by abda11ah** [comment]
I just found this code (provided by perplexity.ai), i will test it tonight :

```
<?php

require 'flight/Flight.php';

// Register your routes and handlers
Flight::route('/', function(){
    echo 'Hello from FlightPHP!';
});

// Worker handler
$handler = static function () {
    Flight::start();
};

// Worker loop
$maxRequests = (int)($_SERVER['MAX_REQUESTS'] ?? 0);
for ($nbRequests = 0; !$maxRequests || $nbRequests < $maxRequests; ++$nbRequests) {
    $keepRunning = \frankenphp_handle_request($handler);
    gc_collect_cycles();
    if (!$keepRunning) break;
}
```

**Comment by n0nag0n** [comment]
You'll probably want to peek at this here. https://github.com/flightphp/core/discussions/519 Yes it is possible, but currently there is a memory leak in the Flight code somewhere and debugging it is really tough. It will still work, you'll just have to restart the process more often than you should cause the memory load and latency will just build up over time.

**Comment by abda11ah** [comment]
Okay, that's very interesting.
But is FlightPHP now stable with Swoole server?

**Reply by n0nag0n** [reply]
I have not made it stable with swoole. I haven't put much effort into the swoole part of things because it really doesn't apply to 99% of developers needs. Most projects are total overkill if they want to use swoole. You need to at a minimum be running hundreds of requests per second AND you need to not have other bottlenecks (like a database) in order for swoole to make sense.

**Comment by abda11ah** [comment]
What do you think of slimframework?
Do you think it could be stable with frankenphp's worker mode?
Or do you know of a framework that's stable in this mode ?

**Reply by n0nag0n** [reply]
I've used slim and I'm sure it's more compatible with FrankenPHP's worker mode. There's a lot of frameworks that work with swoole, roadrunner, frankenphp and others. 

Here's a bunch https://www.techempower.com/benchmarks/#hw=ph&test=fortune&section=data-r22&l=zik073-cn3

And here's a bunch more https://web-frameworks-benchmark.netlify.app/result?l=php

I've worked with Fomo actually and I really do like it. Feels "flightish" to me.

**Reply by abda11ah** [reply]
I'm currently testing docker and workerman. I would like an ultra-fast framework that can manage websockets to try to create a real-time chat. I already have a chat but it works with a check of new messages every 2 seconds, which overload the SQL server with useless requests.....

**Comment by abda11ah** [comment]
Fomo looks good, but it looks like an abandoned project.
And workerman is too low-level and complicated to set up. Just the phpinfo() outputs in text mode because php is run in cli mode..... :(

**Reply by n0nag0n** [reply]
It’s not abandoned. It’s like flight, it’s just kept simple is all. I talked with the maintainer just a couple weeks ago
